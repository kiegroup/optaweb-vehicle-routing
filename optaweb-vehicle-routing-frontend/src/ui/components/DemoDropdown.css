/* Override dropdown menu z-index to be higher than Leaflet map's z-index */
.pf-c-dropdown__menu {
  --pf-c-dropdown__menu--ZIndex: 2000;
}
