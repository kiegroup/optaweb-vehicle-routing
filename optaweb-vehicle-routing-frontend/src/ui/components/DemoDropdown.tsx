/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Dropdown, DropdownItem, DropdownPosition, DropdownToggle } from '@patternfly/react-core';
import * as React from 'react';
import './DemoDropdown.css';

export interface Props {
  demos: string[];
  onSelect: (name: string) => void;
}

const dropdownItems = (demos: string[]): React.ReactNode[] => demos.map(value => (
  <DropdownItem key={value}>
    {value}
  </DropdownItem>
));

export const DemoDropdown: React.FC<Props> = ({ demos, onSelect }) => {
  const [isOpen, setOpen] = React.useState(false);
  return (
    <Dropdown
      style={{ marginBottom: 16, marginLeft: 16 }}
      position={DropdownPosition.right}
      isOpen={isOpen}
      dropdownItems={dropdownItems(demos)}
      onSelect={(e) => {
        setOpen(false);
        onSelect(e.currentTarget.innerText);
      }}
      toggle={(
        <DropdownToggle
          disabled={demos.length === 0}
          onToggle={() => setOpen(!isOpen)}
        >
          Load demo...
        </DropdownToggle>
      )}
    />
  );
};
