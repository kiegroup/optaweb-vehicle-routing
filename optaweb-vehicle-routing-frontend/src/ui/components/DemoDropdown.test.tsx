import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import { DemoDropdown, Props } from './DemoDropdown';

describe('Demo dropdown button', () => {
  it('should render correctly with a couple of demos', () => {
    const props: Props = {
      demos: ['demo 1', 'demo 2'],
      onSelect: jest.fn(),
    };
    const dropdown = shallow(<DemoDropdown {...props} />);
    expect(toJson(dropdown)).toMatchSnapshot();
  });

  it('should be disabled with empty demos', () => {
    const props: Props = {
      demos: [],
      onSelect: jest.fn(),
    };
    const dropdown = shallow(<DemoDropdown {...props} />);
    expect(toJson(dropdown)).toMatchSnapshot();
  });
});
