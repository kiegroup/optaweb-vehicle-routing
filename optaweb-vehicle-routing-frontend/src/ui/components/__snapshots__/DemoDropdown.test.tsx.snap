// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demo dropdown button should be disabled with empty demos 1`] = `
<Dropdown
  dropdownItems={Array []}
  isOpen={false}
  onSelect={[Function]}
  position="right"
  style={
    Object {
      "marginBottom": 16,
      "marginLeft": 16,
    }
  }
  toggle={
    <DropdownToggle
      disabled={true}
      onToggle={[Function]}
    >
      Load demo...
    </DropdownToggle>
  }
/>
`;

exports[`Demo dropdown button should render correctly with a couple of demos 1`] = `
<Dropdown
  dropdownItems={
    Array [
      <DropdownItem>
        demo 1
      </DropdownItem>,
      <DropdownItem>
        demo 2
      </DropdownItem>,
    ]
  }
  isOpen={false}
  onSelect={[Function]}
  position="right"
  style={
    Object {
      "marginBottom": 16,
      "marginLeft": 16,
    }
  }
  toggle={
    <DropdownToggle
      disabled={false}
      onToggle={[Function]}
    >
      Load demo...
    </DropdownToggle>
  }
/>
`;
